### PR TITLE
fix: ensure topic name matches idiomatic variable name

### LINKED ISSUE #1919: Reconsider topic names as variable names
When defining a topic, this is the code:
`var VarName = ftl.Topic[EventType]("name")`
When we export a topic, this is the code we generate
`var Name = ftl.Topic[EventType]("name")`
I think this breaks the mental model people have when switching between modules as the variable they exported in one module is not the name to access it in another.

Also:
- We need to check for name collisions wit the var name with other decls
- We need to make sure the var name starts with a capital letter

### PR DESCRIPTION
**Pull request by safeer**
Fixes #1919

### CODE REVIEW

**Comment by safeer** [comment]
WIP: just need to go from `test_topic` to `TestTopic`

File: go-runtime/compile/schema.go [hunk 1]
```diff
@@ -281,6 +281,13 @@ func extractTopicDecl(pctx *parseContext, node *ast.CallExpr, stack []ast.Node)
 	}
 	topicVarPos := goPosToSchemaPos(varDecl.Specs[0].Pos())
 
+	if t, ok := varDecl.Specs[0].(*ast.ValueSpec); ok {
+		varName := t.Names[0].Name
+		if expVarName := strcase.ToUpperCamel(name); varName != expVarName {
+			pctx.errors.add(errorf(node, "unexpected topic variable name %q, did you mean %q?", varName, expVarName))
```
Comment (matt2e) [inline go-runtime/compile/schema.go:287]: nice

**Review by matt2e — APPROVED** [review]

**Review by worstell — APPROVED** [review]

**Comment by wesbillman** [comment]
@safeer is this one ready to merge or do you have more changes coming?

**Comment by safeer** [comment]
> @safeer is this one ready to merge or do you have more changes coming?

@wesbillman this is a breaking change that requires clients to update their topic varnames

**Comment by safeer** [comment]
Introduced a stripped upper camel case to remove the grossness in the generated code that was being forced upon the original source. See my convo with myself in #1919 😆 

Still breaks client code; varname for a topic like `"public_broadcast"` must now be `var PublicBroadcast`

**Comment by safeer** [comment]
Now prescriptive about topic names. Breaks client code, so not merging yet.

**Pull request merged**